### PR TITLE
LF: parser for LF versions

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
Redesign and move the one defined in `DamlLfEncoder.scala`, so it can be user by third party.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by cocreature — APPROVED** [review]
Thanks! Worth adding a test for this somewhere?

**Review by sofiafaro-da — APPROVED** [review]
Thanks Rémy!

File: daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/LanguageVersion.scala [hunk 1]
```diff
@@ -1,9 +1,8 @@
 // Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-package com.daml.lf.language
-
-import com.daml.lf.VersionRange
+package com.daml.lf
+package language
```
Comment (sofiafaro-da) [inline daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/LanguageVersion.scala:5]: Scala question: What does it mean to have multiple package statements? What's the purpose?

Comment (cocreature) [reply daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/LanguageVersion.scala:5]: it means you can reference `com.daml.lf.X` without qualifying as well as `com.daml.lf.language.X`.

Comment (sofiafaro-da) [reply daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/LanguageVersion.scala:5]: Thanks! :-)

**Pull request merged**